### PR TITLE
Returns missing answers

### PR DESCRIPTION
**Pull request by lmcewanhmrc**
1. The deduction pages handle more missing user answers
2. Big cya directs to account page is claim-credit answer is missing
3. Remove credits feature flag

### CODE REVIEW

File: app/controllers/helpers/NonExportedAmountHelper.scala [hunk 1]
```diff
@@ -49,10 +49,12 @@ class NonExportedAmountHelper { //todo rename?
   }
 
   def getAmountAndDirectlyExportedAnswer(userAnswers: UserAnswers): Option[(Long, Boolean, Boolean)] = {
-    val value = nonExportedAmount(userAnswers).get
-    val isDirectExportYesNo = userAnswers.get(DirectlyExportedPage).get
-    val isAnotherBusinessYesNo = userAnswers.get(AnotherBusinessExportedPage).getOrElse(false) // default to no if not answered
-    Some(value, isDirectExportYesNo, isAnotherBusinessYesNo)
+    for {
```
Comment (lmcewanhmrc) [inline app/controllers/helpers/NonExportedAmountHelper.scala:52]: createResponse needs sorting

File: app/controllers/returns/ImportedPlasticPackagingWeightController.scala [hunk 2]
```diff
@@ -46,27 +46,19 @@ class ImportedPlasticPackagingWeightController @Inject() (
     extends FrontendBaseController with I18nSupport {
 
   def onPageLoad(mode: Mode): Action[AnyContent] =
-    (identify andThen getData andThen requireData).async {
+    (identify andThen getData andThen requireData) {
```
Comment (lmcewanhmrc) [inline app/controllers/returns/ImportedPlasticPackagingWeightController.scala:49]: journey action

File: app/controllers/returns/ReturnsCheckYourAnswersController.scala [hunk 3]
```diff
@@ -83,37 +83,47 @@ class ReturnsCheckYourAnswersController @Inject()(
...
   }
 
-  private def getCredits(request: DataRequest[_], isFirstReturn: Boolean)
-                        (implicit hc: HeaderCarrier, messages: Messages): Future[Either[ServiceError, Credits]] =
-    if (isFirstReturn || !appConfig.isCreditsForReturnsFeatureEnabled)
+  private def callCalculationAndCreditApi(request: DataRequest[_], isUserClaimingCredit: Boolean, isFirstReturn: Boolean) 
+    (implicit messages: Messages, hc: HeaderCarrier): Future[(Calculations, Credits)] = {
+    
+    for {
+      calculations <- returnsConnector.getCalculationReturns(request.pptReference)
```
Comment (lmcewanhmrc) [inline app/controllers/returns/ReturnsCheckYourAnswersController.scala:113]: go async

File: app/controllers/returns/ReturnsCheckYourAnswersController.scala [hunk 4]
```diff
@@ -83,37 +83,47 @@ class ReturnsCheckYourAnswersController @Inject()(
...
-    } yield (calculations, credits) match {
-      case (Right(calculations), Right(credits)) => (calculations, credits)
-      case _ => throw new RuntimeException("Error: There was a problem retrieving return calculation or the credits balance")
+    request.userAnswers.get(WhatDoYouWantToDoPage) match {
+      case Some(isUserClaimingCredit) => displayPage2(request, obligation, isUserClaimingCredit, false) // TODO making an assumption here, is it sound?
+      case None =>
+        taxReturnHelper.nextOpenObligationAndIfFirst(request.pptReference).flatMap {
+          case Some((_, true)) => displayPage2(request, obligation, false, true)
+          case _ => Future.successful(Redirect(controllers.routes.IndexController.onPageLoad))
+        }
```
Comment (lmcewanhmrc) [inline app/controllers/returns/ReturnsCheckYourAnswersController.scala:95]: deal with 3 states better?

File: test/controllers/returns/credits/ConfirmPackagingCreditControllerSpec.scala [hunk 5]
```diff
@@ -17,28 +17,25 @@
 package controllers.returns.credits
 
 import base.utils.JourneyActionAnswer
+import cacheables.ReturnObligationCacheable
```
Comment (lmcewanhmrc) [inline test/controllers/returns/credits/ConfirmPackagingCreditControllerSpec.scala:20]: missing a test for the redirect?

**Review by mauro-fanton — APPROVED** [review]

File: app/connectors/CalculateCreditsConnector.scala [hunk 6]
```diff
@@ -31,12 +31,8 @@ class CalculateCreditsConnector @Inject()(httpClient: HttpClient, appConfig: Fro
 
     val timer = metrics.defaultRegistry.timer("ppt.exportcredits.open.get.timer").time()
```
[the remark this answers is not in the document]
Comment (lmcewanhmrc) [reply app/connectors/CalculateCreditsConnector.scala]: Done

**Pull request merged**